### PR TITLE
Add external topology mode

### PR DESCRIPTION
**Pull request by csrwng**
Implementation of https://github.com/openshift/enhancements/blob/master/enhancements/external-control-plane-topology.md
Adds External value to the ToplogyMode type.

### CODE REVIEW

File: config/v1/0000_10_config-operator_01_infrastructure.crd.yaml [hunk 1]
```diff
@@ -142,6 +143,7 @@ spec:
                   enum:
                     - HighlyAvailable
                     - SingleReplica
+                    - External
```
Comment (bparees) [inline config/v1/0000_10_config-operator_01_infrastructure.crd.yaml:146]: are we actually consistent about infra?  

in hypershift we have some infra components that run external (olm) and some that run on cluster (registry) right?

Comment (csrwng) [reply config/v1/0000_10_config-operator_01_infrastructure.crd.yaml:146]: That is correct. From an operator point of view, I don't think the External value makes sense for infra topology (Unless we really externalize all of infra). It is meant to apply only to control plane topology for now. However, not sure that we can express that through the API as it is.

Comment (bparees) [reply config/v1/0000_10_config-operator_01_infrastructure.crd.yaml:146]: So "external" will not be a valid value for status.infrastructureTopology ?

I guess can understand not creating another Type to split them, so i guess we can live with this but it is somewhat unfortunate that it confusingly implies it is valid, even if practice we'd never actually set it to that value.

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -96,7 +98,7 @@ type InfrastructureStatus struct {
 }
 
 // TopologyMode defines the topology mode of the control/infra nodes.
-// +kubebuilder:validation:Enum=HighlyAvailable;SingleReplica
+// +kubebuilder:validation:Enum=HighlyAvailable;SingleReplica;External
```
Comment (dhellmann) [inline config/v1/types_infrastructure.go:101]: If we move this declaration up to the fields above, we can provide different enum values and prevent the infra topology from being set to external. We would want a comment here to explain why we're doing it that way, too.

Comment (csrwng) [reply config/v1/types_infrastructure.go:101]: @dhellmann I like this suggestion. We can then make sure we don't set an invalid value for infra topology. Will submit a follow up PR

**Pull request merged**